NitroxLauncher/Pages/ServerPage.xaml.cs

```diff
@@ -1,37 +1,848 @@
+        private void RBServer_Clicked(object sender, RoutedEventArgs e)
+        {
+            LauncherLogic.Config.IsExternalServer = RBIsExternal.IsChecked ?? true;
+        }
+
+        public void InitializeWorldListing()
+        {
+            WorldManager.Refresh();
+
+            NoWorldsBackground.Opacity = WorldManager.GetSaves().Any() ? 0 : 1;
+
+            // Bind the list data to be used in XAML
+            WorldListingContainer.ItemsSource = null;
+            WorldListingContainer.ItemsSource = WorldManager.GetSaves();
+        }
+
+        // File Management
+        public void SaveConfigSettings()
+        {
+            // If world name was changed, rename save folder to match it
+            string dest = Path.Combine(Path.GetDirectoryName(SelectedWorldDirectory) ?? throw new Exception("Selected world is empty"), TBWorldName.Text);
+            if (SelectedWorldDirectory != dest)
+            {
+                Directory.Move(SelectedWorldDirectory, dest+" temp"); // These two lines are needed to handle names that change in capitalization,
+                Directory.Move(dest+" temp", dest);                   // since Windows still thinks of the two names as the same.
+                SelectedWorldDirectory = dest;
+            }
+
+            Config = ServerConfig.Load(SelectedWorldDirectory);
+            Config.Update(SelectedWorldDirectory, c =>
+            {
+                c.SaveName = TBWorldName.Text;
+                if (IsNewWorld) { c.Seed = TBWorldSeed.Text; }
+                if (RBFreedom.IsChecked == true) { c.GameMode = ServerGameMode.FREEDOM; }
+                else if (RBSurvival.IsChecked == true) { c.GameMode = ServerGameMode.SURVIVAL; }
+                else if (RBCreative.IsChecked == true) { c.GameMode = ServerGameMode.CREATIVE; }
+                else if (RBHardcore.IsChecked == true) { c.GameMode = ServerGameMode.HARDCORE; }
+
+                c.DisableConsole = !(bool)CBCheats.IsChecked;
+                c.MaxConnections = Convert.ToInt32(TBMaxPlayerCap.Text);
+                if (CBBDefaultPerms.SelectedIndex == 0) { c.DefaultPlayerPerm = Perms.PLAYER; }
+                else if (CBBDefaultPerms.SelectedIndex == 1) { c.DefaultPlayerPerm = Perms.MODERATOR; }
+                else if (CBBDefaultPerms.SelectedIndex == 2) { c.DefaultPlayerPerm = Perms.ADMIN; }
+
+                c.CreateFullEntityCache = (bool)CBCreateFullEntityCache.IsChecked;
+                c.DisableAutoSave = !(bool)CBAutoSave.IsChecked;
+                c.AutoPortForward = (bool)CBAutoPortForward.IsChecked;
+                c.SaveInterval = Convert.ToInt32(TBSaveInterval.Text)*1000;  // Convert seconds to milliseconds
+                if ((bool)CBEnableJoinPassword.IsChecked) { c.ServerPassword = TBJoinPassword.Text; }
+                else { c.ServerPassword = string.Empty; }
+                c.ServerPort = Convert.ToInt32(TBWorldServerPort.Text);
+                c.LANDiscoveryEnabled = (bool)CBLanDiscovery.IsChecked;
```

Comment:
These bool casts can throw exceptions if the `IsChecked` is null (see below). It should be changed to have sane fallback values like so:
```cs
c.DisableConsole = !CBCheats.IsChecked ?? false; // aka: if not set, false = enable console
```

```cs
bool? b = null;
> (bool)b
> System.InvalidOperationException: Nullable object must have a value.
  + System.ThrowHelper.ThrowInvalidOperationException(System.ExceptionResource)
  + Nullable<T>.get_Value()
  ```

Reply:
I've updated the fallback logic to use the config value since they have default values defined. This should be a better UX.

Reply:
That's a good plan, I was thinking about doing something like that instead of defining the default values again.